lib/linzer.rb

```diff
@@ -59,7 +59,12 @@ def sign!(request_or_response, **args)
 
     def verify!(request_or_response, key: nil, no_older_than: 900)
       message = Message.new(request_or_response)
-      signature = Signature.build(message.headers.slice("signature", "signature-input"))
+      signature_headers = {}
+      %w[signature-input signature].each do |name|
+        value = message.header(name)
+        signature_headers[name] = value if value
+      end
```

Comment:
Not having a full hash means we lose the convenience of `Hash#slice`. I do not think this is a huge problem, *but* I dislike having to repeat this in the rack auth middleware.

The only alternative I could come up with, was passing the full `Linzer::Message` object to `Signature.build`. Not sure if that is something you would consider. But maybe you have another idea?